### DISCUSSION TITLE
Parser API: pattern match inner pairs as slice

### DISCUSSION THREAD

**Discussion by shishkin** [pest-parser/pest#722] [Q&A] [answered]
Hi all! I'm wondering if it is possible to pattern match `pair`'s inner pairs as a slice? I'm new to Rust and coming from Scala look for more convenience on matching parser results, e.g.:

```rust
match pair {
  Pair(Rule::binary_expression, [left, right]) => {
    // left and right are of type Pair and can be matched further
  }
}
```

I've tried to build my own convenience layer of structs and traits on top, but can't overcome the borrow checker. Any advice appreciated.

**Comment by tomtau** [comment, marked as answer]
`Pair`'s parts are private, so it's not possible to match on them directly. As for "matching parser results convenience", I'd suggest checking out some of the existing crates, such as [pest_consume](https://crates.io/crates/pest_consume) or [pest-ast](https://crates.io/crates/pest-ast).
